test/e2e/certificate_publisher_test.go

```diff
@@ -0,0 +1,168 @@
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	ingresscontroller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// TestCreateIngressControllerThenSecret creates an ingresscontroller that
+// references a secret that does not exist, then creates the secret and verifies
+// that the operator updates the "router-certs" global secret.
+func TestCreateIngressControllerThenSecret(t *testing.T) {
+	cl, ns, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create the ingresscontroller.
+	var one int32 = 1
+	ic := &operatorv1.IngressController{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-ingresscontroller-then-secret",
```

Comment:
Not saying we need it here, but before I forget, could we use GenerateName when unique names are needed?

Reply:
Fixed.